src/components/activePanel/directives/activePanel.directive.js

```diff
@@ -15,61 +15,90 @@
-                                SHARE_MY_SCREEN: translation[translateNamespace + '.' + 'SHARE_MY_SCREEN'],
-                                END_SCREEN_SHARING: translation[translateNamespace + '.' + 'END_SCREEN_SHARING']
+                        const timerSecondInterval = 1000;
+                        const activePanelVisibleClassName = 'activePanel-visible';
+                        const isStudent = ENV.appContext.toLowerCase() === 'student';
+                        const isTeacher = ENV.appContext.toLowerCase() === 'dashboard';
+                        let durationToDisplay;
+                        let timerInterval;
+                        let liveSessionData;
+                        let liveSessionDuration = 0;
+                        let prevLiveSessionStatus = UserLiveSessionStateEnum.NONE.enum;
+
+
+                        activePanelInit();
+
+                        function activePanelInit() {
+                            getTranslations();
+
+                            scope.d = {
+                                states: {
+                                    NONE: 0,
+                                    LIVE_SESSION: 1
+                                },
+                                shareScreenBtnsEnable: true,
+                                disableAllBtns: false,
+                                isTeacher: isTeacher,
+                                presenceStatusMap: PresenceService.userStatus,
+                                endScreenSharing: endScreenSharing,
+                                endSession: endLiveSession,
+                                viewOtherUserScreen: viewOtherUserScreen,
+                                shareMyScreen: shareMyScreen,
+                                openHangouts: openHangouts
                             };
-                        }).catch(function (err) {
-                            $log.debug('Could not fetch translation', err);
-                        });
 
-                        UserProfileService.getProfile().then(function (userProfile) {
-                            scope.d.userProfile = userProfile;
-                        });
+                            UserProfileService.getProfile().then(userProfile => scope.d.userProfile = userProfile);
 
-                        function endLiveSession() {
-                            function endSessionAndStopTrack(liveSessionData) {
-                                deleteStudentPath(liveSessionData.studentId);
-                                LiveSessionSrv.endLiveSession(liveSessionData.guid);
-                                PresenceService.stopTrackUserPresence(liveSessionData.studentId);
-                                PresenceService.stopTrackUserPresence(liveSessionData.educatorId);
-                            }
+                            ScreenSharingSrv.registerToCurrUserScreenSharingStateChanges(listenToScreenShareStatus);
```

Comment:
There is no need to wait for UserProfileService.getProfile() ?

Reply:
No, the fn who use it, check if the userProfile is exist on the scope